### PR TITLE
Add framegrabber recipe

### PR DESCRIPTION
**Pull request by TSAnsari**
This recipe will build and install framegrabber kernel
module on arp.

Signed-off-by: tansari <tansari@luxoft.com>

### CODE REVIEW

File: conf/machine/arp.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#@TYPE: Machine
+#@NAME: arp
+
+#@DESCRIPTION: Machine configuration for Automotive Reference Platform
+
+require conf/machine/intel-corei7-64.conf
```
Comment (thenor) [inline conf/machine/arp.conf:6]: Do we have this conf file?

Comment (TSAnsari) [reply conf/machine/arp.conf:6]: Yes this is part of meta-intel layer

Comment (thenor) [reply conf/machine/arp.conf:6]: Alright.

**Comment by sashko** [comment]
Add framegrabber recipe / Add recipe for framegrabber

File: recipes-kernel/framegrabber/framegrabber_git.bb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+SUMMARY = "Framegrabber kernel module"
+DESCRIPTION = "Kernel module to load video frames captured from ARP cameras in to memory"
```
Comment (sashko) [inline recipes-kernel/framegrabber/framegrabber_git.bb]: into

File: recipes-kernel/framegrabber/framegrabber_git.bb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+inherit module
+
+SRCREV = "e6ac6f58abee88d04d7c92785c2c9a6c3868aa48"
+
+SRC_URI = "git://github.com/Pelagicore/framegrabber.git;protocol=https;branch=master"
+
+S = "${WORKDIR}/git"
+
+# The inherit of module.bbclass will automatically name module packages with
```
Comment (sashko) [inline recipes-kernel/framegrabber/framegrabber_git.bb]: why is this needed?

Comment (TSAnsari) [reply recipes-kernel/framegrabber/framegrabber_git.bb]: So it fetches the sources in the S directory to compile and install

Comment (sashko) [reply recipes-kernel/framegrabber/framegrabber_git.bb]: I've meant the comment.

Comment (TSAnsari) [reply recipes-kernel/framegrabber/framegrabber_git.bb]: Its not needed. The comment was there from where I fetched this recipe. So I didn't removed this. Do you think this is not a useful comment?

Comment (sashko) [reply recipes-kernel/framegrabber/framegrabber_git.bb]: I thought you were the author; it's not important

**Review by sashko — CHANGES_REQUESTED** [review]
Fix the commit message, please

File: recipes-kernel/framegrabber/framegrabber_git.bb (hunk 3 above)
Comment (frznlogic) [reply recipes-kernel/framegrabber/framegrabber_git.bb]: Tbh, I kind of agree, at the very least the comment is in the wrong location in the file imho. It's a dangling comment.

File: conf/machine/arp.conf (hunk 1 above)
Comment (frznlogic) [reply conf/machine/arp.conf:6]: Is the ARP only usable with intel-corei7-64? Does it make sense to abstract/split this in some way?

**Review by frznlogic — COMMENTED** [review]
LGTM except for the two comments I added.

File: conf/machine/arp.conf (hunk 1 above)
Comment (TSAnsari) [reply conf/machine/arp.conf:6]: As far as I understood, ARP will only support intel cpu. May be its too premature to abstract it at the moment when we don't know if there will be other CPUs for ARP.

**Comment by sashko** [comment]
> Add framegrabber recipe to meta-arp layer

This **is** the 'meta-arp' layer. Why do you need to specify this?

File: conf/machine/arp.conf (hunk 1 above)
Comment (FlorentRevest) [reply conf/machine/arp.conf:6]: @frznlogic we kinda came to an agreement with @GordanM that this abstraction should be kept:
- 1st, for the future of meta-arp
- 2nd, as a reference implementation of a BSP

Comment (thenor) [reply conf/machine/arp.conf:6]: Alright. Sounds good to me then.

**Review by thenor — APPROVED** [review]

**Review by frznlogic — APPROVED** [review]
